var/spack/repos/builtin/packages/zfp/package.py

```diff
@@ -56,15 +59,25 @@ def edit(self, spec, prefix):
             spec.variants['bsws'].value)
 
     def build(self, spec, prefix):
-        make("shared")
+        with working_dir('src'):
+            if '~shared' in spec:
+                make('static')
+            else:
+                make('shared')
 
     def install(self, spec, prefix):
         incdir = 'include' if spec.satisfies('@0.5.1:') else 'inc'
 
+        if self.run_tests:
+            make('test')
```

Comment:
You shouldn't need to add this, it's already in the MakefilePackage base class.

Reply:
Really? Hmm. Doesn't that mean that MakefilePackage base class assumes all such packages implement a `test` target. I can name several that don't. I'm confused.

Reply:
no, it does not assume this. It calls it if the target is defined

Reply:
Hmm. How does Spack discover if the target is defined?

Reply:
See https://github.com/LLNL/spack/blob/develop/lib/spack/spack/build_systems/makefile.py#L111,L117 and https://github.com/LLNL/spack/blob/develop/lib/spack/spack/package.py#L1083,L1102

Reply:
Wow. Cool...so this is a case where Spack is doing some work to discover properties of a package. And, its even considering multiple possible alternatives in this case, `test` or `check`.

Reply:
Yep and also check out spack create, where it's digging into the archive to figure out which build system to use.  @adamjstewart did a lot of work on that.